@here/harp-mapview/lib/Utils.ts

```diff
@@ -26,29 +26,30 @@ const groundNormalPlanarProj = new THREE.Vector3(0, 0, 1);
 const groundNormalInvPlanarProj = groundNormalPlanarProj.clone().negate();
 const groundPlane = new THREE.Plane(groundNormalPlanarProj.clone());
 const groundSphere = new THREE.Sphere(undefined, EarthConstants.EQUATORIAL_RADIUS);
-const cameraZPosition = new THREE.Vector3(0, 0, 0);
-const rotationMatrix = new THREE.Matrix4();
-const unprojectionMatrix = new THREE.Matrix4();
 const rayCaster = new THREE.Raycaster();
-const yawQuaternion = new THREE.Quaternion();
-const pitchQuaternion = new THREE.Quaternion();
-const tmpQuaternion = new THREE.Quaternion();
-const tmpMatrix = new THREE.Matrix4();
 const tmpVectors: THREE.Vector3[] = [new THREE.Vector3(), new THREE.Vector3(), new THREE.Vector3()];
 
-export namespace MapViewUtils {
-    /**
-     * The yaw axis around we rotate when we change the yaw.
-     * This axis is fix and is the -Z axis `(0,0,1)`.
-     */
-    const yawAxis = new THREE.Vector3(0, 0, 1);
-
-    /**
-     * The pitch axis which we use to rotate around when we change the pitch.
-     * The axis is fix and is the +X axis `(1,0,0)`.
-     */
-    const pitchAxis = new THREE.Vector3(1, 0, 0);
+/**
+ * Cached ThreeJS instances for realtime maths.
+ */
+const space = {
+    x: new THREE.Vector3(),
+    y: new THREE.Vector3(),
+    z: new THREE.Vector3()
+};
+const tangentSpace = {
```

Comment:
Doesn't it make more sense to use [`Matrix3`](https://threejs.org/docs/#api/en/math/Matrix3) here?

Reply:
Agree, though see other comment.